bin/anthology/papers.py

```diff
@@ -187,18 +136,26 @@ def _interpret_pages(self):
                 return
 
     @property
-    def is_volume(self):
-        """Determines if this paper is a regular paper or a proceedings volume.
+    def collection_id(self):
+        return self.parent_volume.collection_id
 
-        By default, each paper ID of format 'x000' will be treated as (the front
-        matter of) a proceedings volume, unless the XML is of type workshop,
-        where each paper ID of format 'xx00' is treated as one volume.
-        """
-        return is_volume_id(self.full_id)
+    @property
+    def volume_id(self):
+        return self.parent_volume.volume_id
+
+    @property
+    def paper_id(self):
+        if self.collection_id[0] == "W" or self.collection_id == "C69":
+            # If volume is a workshop, use the last two digits of ID
+            _id = "{}{:02d}".format(self.volume_id, int(self._id))
+        else:
+            # If not, only the last three
+            _id = "{}{:03d}".format(self.volume_id, int(self._id))
+        return _id
```

Comment:
When IDs can now have a variable number of digits, let's add an `assert len(_id) == 4` or something just to be sure.

Reply:
Good idea.